client/src/components/CreatePost.tsx

```diff
@@ -8,173 +8,251 @@ import TextField from "@mui/material/TextField";
 import Divider from "@mui/material/Divider";
 import Typography from "@mui/material/Typography";
 import Tooltip from "@mui/material/Tooltip";
-
+import Dialog from "@mui/material/Dialog";
+import { ArrowBack } from "@mui/icons-material";
 import PreviewPopUp from "./PreviewPopUp";
+import Snackbar from "@mui/material/Snackbar";
+
+import ServerApi from "../api/v1/index";
+
+const API = new ServerApi();
 
 function CreatePost() {
-  // create hooks for input fields
+  // hooks
   const [form, setForm] = useState({
     title: "",
     body: "",
     file: "",
     tags: "",
-    eventCapacity: "",
-    eventLocation: "",
+    capacity: "",
+    location: "",
   });
+  const [openPopup, setOpenPopup] = useState(false); // for preview popup
+  const [isAlertOpen, showAlert] = useState(false); // for snackbar
+  const [alertMsg, setMsg] = useState("An error has occurred"); // for snackbar message
+  const [capacityError, setCapacityError] = React.useState(""); // for capacity input validation
+  const [isOpen, toggleDialog] = useState(false); // for create post dialog toggle
 
-  const [openPopup, setOpenPopup] = useState(false);
-
-  // handle functions
+  const closeDialog = () => {
+    toggleDialog(false);
+  };
   const handleClickOpen = () => {
     setOpenPopup(true);
   };
 
-  // handle functions
+  // handling
+  const handleSubmit = () => {
+    API.createPost(`/posts/`, form)
+      .then((res: any) => {
+        if (res.response.status === 204) {
+          setMsg("Post has been succesfully created.");
+        } else {
+          setMsg("Failed to create post");
+        }
+        showAlert(true);
+      })
+      .catch((err) => {
+        console.error(err);
+        setMsg("Failed to create post. Ensure all the fields are correct");
+        showAlert(true);
```

Comment:
since show alert is executed in both cases, you can put it into a `finally`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally